Registry Refactor

Klebestreifen opened the pull request.

Klebestreifen committed:
[WIP] start registry refactor

MasterEnderman committed:
refactored the block registry
i started to refactor all classes that mention the old registry only thing left for now is the RecipeRegistry.

MasterEnderman commented on src/main/java/teamroots/embers/register/BlockRegister.java:
```diff
@@ -2,184 +2,373 @@
+    public static final Block BLOCK_ASTRALITE = new BlockBase(Material.ROCK, "block_astralite", true).setHarvestProperties("pickaxe", 2).setHardness(1.6f).setLightOpacity(16).setCreativeTab(Embers.resource_tab).setCreativeTab(Embers.resource_tab);
+    public static final Block BLOCK_BRONZE = new BlockBase(Material.ROCK, "block_bronze", true).setBeaconBase(true).setHarvestProperties("pickaxe", 1).setHardness(2.3f).setLightOpacity(16).setCreativeTab(Embers.resource_tab);
+    public static final Block BLOCK_CAMINITE_BRICK = new BlockBase(Material.ROCK, "block_caminite_brick", true).setHarvestProperties("pickaxe", 0).setHardness(1.6f).setLightOpacity(16);
+    public static final Block BLOCK_CAMINITE_BRICK_SLAB = new BlockSlabBase(BLOCK_CAMINITE_BRICK_SLAB_DOUBLE, "block_caminite_brick_slab", true).setHarvestProperties("pickaxe", 0).setIsFullCube(false).setIsOpaqueCube(false).setHardness(1.6f).setLightOpacity(1);
```
@Klebestreifen this is the problem with the slabs I mentioned before if you want to try to figure out how to avoid this.

MasterEnderman commented on src/main/java/teamroots/embers/register/BlockRegister.java:
```diff
@@ -2,184 +2,373 @@
 
     ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
     private final static List<Block> ALWAYS_REGISTER = Collections.singletonList(
```
@Klebestreifen IntelliJ seems to have a problem with the singleton list as it want's to remove all list entries and I have no idea why, maybe you can help me figuring that out.

MasterEnderman reviewed (commented):
First iteration of the block registry refactor. There are still quite a few things we need to figure out.

MasterEnderman committed:
updated BlockRegister

MasterEnderman committed:
added ArmorMaterialRegister

MasterEnderman committed:
moved Entities to its own register

MasterEnderman committed:
moved Fluids to its own register

MasterEnderman committed:
moved ItemModifiers to its own register

MasterEnderman committed:
added MaterialRegister

MasterEnderman committed:
added ToolMaterialRegister

MasterEnderman committed:
extracted Items to its own register

MasterEnderman committed:
Refactor for new RegisterManagement

MasterEnderman committed:
Added vintagefix and loliASM to dev-env

MasterEnderman commented:
Almost done with the refactor, but right now I encounter an error where the game crashes as soon as I pick up any bauble in HEI. Here is the crash report: https://mclo.gs/FqPUjpq
If anyone got an idea what could be the cause for it feel free to let me know.

Klebestreifen commented:
The problem comes with a GroovyScript mixin. [Here](https://github.com/CleanroomMC/GroovyScript/blob/6592983be2bbbab916fd80e9d8e08b1ed7be1572/src/main/java/com/cleanroommc/groovyscript/core/mixin/ItemStackMixin.java#L134C72-L134C85)
I suspect it is due to the different mappings used. It should therefore work in obfuscated mode. Otherwise ask the Claenroom team.

MasterEnderman committed:
moved to cleanroom maven updated build.gradle

MasterEnderman commented:
Alright I sorted out the crash when interacting with a bauble. Now it's back to fix the dependency issue and finish the registry refactor.

MasterEnderman committed:
made Embers independend from Baubles and Mystical Mechanics, more registry refacor and fixed some missing localizations

MasterEnderman committed:
finished registry refactor
I know that some of them are a bit scuffed, and they can probably be improved further, but I would like to start on adding more config options while integrating the first few addons next. I guess things some just need to 'work' for now.

MasterEnderman commented:
I would call this part of the refactor 'finished' for now, it may be a bit scuffed in some places, but managing blocks and items is now much easier to navigate and extend, when we start integrating various addons. If there's nothing you want to change yourself, feel free to merge this back into the config branch, so I can start on adding more config options.
